requirements.txt

```diff
@@ -8,4 +8,5 @@ seaborn
 numba
 isort
 flake8
-flake8-isort
\ No newline at end of file
+flake8-isort
+cupy-cuda12x
```

Comment:
We cannot merge the code as-it-is, as installation of `cupy-cuda12x` doesn't make sense for CPU-only users. We need either some kind of smart dispatcher or a smarter way to manage deps.